elastic/elastic-agent
# Send upgrade details to Fleet Server in check-in API requests

## What does this PR do?

This PR sends the current upgrade details to Fleet Server as part of the check-in API's request body.

## Why is it important?



To give Fleet users more visibility into the Agent upgrade process.

## Checklist





## How to test this PR locally

1. Spin up a stack deployment with the latest SNAPSHOT version.  As of this writing, that would be version `8.12.0-SNAPSHOT`.
2. Build Elastic Agent from this PR's branch but deliberately giving it a version older than the latest SNAPSHOT version.
   ```
   AGENT_PACKAGE_VERSION=8.3.0 EXTERNAL=true SNAPSHOT=true PLATFORMS=darwin/arm64 PACKAGES=targz mage package
   ```
3. Enroll the built Agent with Fleet.
4. Once the Agent shows up in the Fleet UI, upgrade it to the latest SNAPSHOT version available. For this, you will have to manually enter the version in the "Upgrade version" text box in the popup, e.g. `8.12.0-SNAPSHOT`.
5. Observe that the Fleet UI shows upgrade details, for example:
   <img width="1588" alt="Screenshot 2023-11-08 at 14 18 10" src="https://github.com/elastic/elastic-agent/assets/51061/58a8f0fc-a058-49b7-80d5-776fedf7839b">

## Related issues


- Relates to https://github.com/elastic/elastic-agent/issues/3119#issuecomment-1743638526
